SteamFriends and SteamUser Changes

Crytilis opened the pull request:
This PR would implement the following changes as listed on the commits;

- Added AdvertiseGame to SteamUser so that it may be used for clients.
- PersonaChange enum is now set to public rather than internal so it may used with the PersonaStateChange callback.
- Modified the PersonaStateChange callback for SteamFriends to include the PersonaStateChange_t flag as defined in the PersonaChange enum.
- Removed the F word from the readme in the event that prospective employers are looking at a forked GitHub.

garrynewman commented on Facepunch.Steamworks/Generated/SteamEnums.cs:
```diff
@@ -546,7 +546,7 @@ internal enum ActivateGameOverlayToWebPageMode : int
 	// EPersonaChange
 	//
-	internal enum PersonaChange : int
+	public enum PersonaChange : int
```
This needs to be updated in the generator too - not just here.

Crytilis replied:
I had overlooked this by mistake, corrections made as per my latest comment.

Crytilis commented:
Must have overlooked that, the correction has been made. I did make a few more changes some of which were for compilation testing. 

- Updated Test Adaptor and Framework packages
- Updated Newtonsoft packages
- Modified the folder path to account for .NET6 changes
- Set the steam_api.json file to copy to output upon compilation to prevent compile errors.

garrynewman commented:
There's a few too many unexplained changes here for me to pull this.
